Examples/MultiModal/Mixed_Image__And_Tabular_Reading/.ipynb_checkpoints/CompositeReader-checkpoint.ipynb

```diff
@@ -0,0 +1,429 @@
+    "        eval_error = trainer.previous_minibatch_evaluation_average\n",
+    "        avg_out_error = calc_avg_per_out_error(eval_error, num_outputs)\n",
+    "        print (\"Minibatch: {0}, Loss: {1:.4f}, +/- error per output: {2:.4f}\".format(mb, training_loss, avg_out_error))\n",
+    "        \n",
+    "    return mb, training_loss, eval_error"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configure Train and Test Loop\n",
+    "\n",
+    "This is pretty much boiler plate as well, but I would suggest reading some of the other articles that really focus on this area because there are some great optimizations you could make here that I just left out for the sake of simplicity.\n",
+    "\n",
+    "**The big things to notice here are: ** the input map is very important.  input_map and test_input_map.  Notice they pull out those specific streamdef variables we defined in our reader and populate our placeholder variables y, x_i and x_t.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def train_test(train_reader, test_reader, num_outputs, num_sweeps_to_train_with=10):\n",
+    "    \n",
+    "    # Instantiate the loss and error function\n",
+    "    # z comes from a global scope outside of this function (defined later).\n",
+    "    loss, label_error = create_errors(z, y)\n",
+    "    \n",
+    "    # Instantiate the trainer object to drive the model training\n",
+    "    # we use a super low learning rate so we don't get exploding gradients\n",
+    "    # which is common for regressors in large networks.\n",
+    "    learning_rate = .00001 \n",
+    "    lr_schedule = C.learning_rate_schedule(learning_rate, C.UnitType.minibatch)\n",
+    "    learner = C.sgd(z.parameters, lr_schedule)\n",
+    "    trainer = C.Trainer(z, (loss, label_error), [learner])\n",
+    "    \n",
+    "    # Initialize the parameters for the trainer\n",
+    "    minibatch_size = 64\n",
+    "    num_samples_per_sweep = 60000\n",
+    "    num_minibatches_to_train = (num_samples_per_sweep * num_sweeps_to_train_with) / minibatch_size\n",
+    "    \n",
+    "    # Map the data streams to the input and labels.\n",
+    "    # this is where we can pull in our label pairs\n",
+    "    input_map={\n",
+    "        y  : train_reader.streams.labels,\n",
+    "        x_i  : train_reader.streams.features_image,\n",
+    "        x_t  : train_reader.streams.features_tabular\n",
+    "    } \n",
+    "    \n",
+    "    training_progress_output_freq = 500\n",
+    "     \n",
+    "    # Start a timer\n",
+    "    start = time.time()\n",
+    "\n",
+    "    for i in range(0, int(num_minibatches_to_train)):\n",
+    "        # Read a mini batch from the training data file\n",
+    "        data=train_reader.next_minibatch(minibatch_size, input_map=input_map)         \n",
```

Comment:
can we please use training_session: refer to https://github.com/Microsoft/CNTK/blob/master/Manual/Manual_How_to_feed_data.ipynb